smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -121,8 +128,18 @@ contract MixinTransfer is
     public
   {
     _isValidKey(_tokenId);
-    _onlyKeyManagerOrApproved(_tokenId);
-    if(ownerOf(_tokenId) != _from) {
+    if(
```

Comment:
That if statement is a bit confusing. Correct me if I am wrong but the goal here is to only allow this if:
* the msg.sender is both _from AND the ownerOf(_tokenId) ANd _tokenId does not have a manager 
OR
* the msg.sender is the the keyManager of tokenId
OR
* the msg.sender has been approved for the transfer

If so, would something like this be more readable:
```
if (!
    (msg.sender == _from && ownerOf(_tokenId) == _from && approved[_tokenId] == 0x) OR
    msg.sender == approved[_tokenId] OR
    approved[_tokenId] == msg.sender OR
    isApprovedForAll(_ownerOf[_tokenId], msg.sender)
)
```

Reply:
oh that want what I had in mind. I thought in any case if a key manager was set then the transfer will be made impossible, regardless if the transfer was initiated by the key owner, the key manager or an approved address.

Reply:
ok in that scenario, can we make sure that a key manager can "renounce" that role?